### PR TITLE
Make comment ID a simple `string`

### PR DESCRIPTION
**Pull request by mxdvl**
## What does this change?

Only consider comment IDs to be strings.

## Why?

The value was being flip-flopped between `string` and `number`, but we only really need it to be a `string`, and that is how it comes to us from the API responses.

See https://github.com/guardian/dotcom-rendering/pull/11081#discussion_r1555013650 and https://github.com/guardian/bridget/pull/148#discussion_r1555533198

Stop coercing primitives to what they do not want to be! 💪

## Screenshots

N/A

### CODE REVIEW

File: dotcom-rendering/src/components/Discussion/Comments.stories.tsx [hunk 1]
```diff
@@ -1,20 +1,31 @@
...
 export default { component: Comments, title: 'Discussion/App' };
 
+const discussionMock = parse(discussionApiResponseSchema, discussion);
+assert(discussionMock.status === 'ok');
+const discussionWithTwoComments = parse(
+	discussionApiResponseSchema,
+	discussionWithTwoCommentsMock,
+);
+assert(discussionWithTwoComments.status === 'ok');
+
```
Comment (mxdvl) [inline dotcom-rendering/src/components/Discussion/Comments.stories.tsx:28]: Not sure how we could make this cleaner… we’re parsing the mock fixture to match the internatl representation.

File: dotcom-rendering/src/components/Discussion.tsx [hunk 2]
```diff
@@ -83,10 +83,9 @@ const commentIdFromUrl = () => {
  */
 const remapToValidFilters = (
 	filters: FilterOptions,
-	hashCommentId: number | undefined,
+	hashCommentId: string | undefined,
 ) => {
-	const permalinkBeingUsed =
-		hashCommentId !== undefined && !Number.isNaN(hashCommentId);
+	const permalinkBeingUsed = !!hashCommentId;
```
Comment (abeddow91) [inline dotcom-rendering/src/components/Discussion.tsx:88]: 😍

**Review by abeddow91 — APPROVED** [review]
Nice, it much cleaner now we're not coercing the value a bunch of places - thanks for refactoring!

**Review by georgeblahblah — APPROVED** [review]
🧵

File: dotcom-rendering/src/lib/discussion.ts [hunk 3]
```diff
@@ -75,7 +74,7 @@ export interface UserProfile {
 }
 
 const baseCommentSchema = object({
-	id: number(),
+	id: transform(union([number(), string()]), (id) => id.toString()),
```
Comment (mxdvl) [inline dotcom-rendering/src/lib/discussion.ts:77]: We’re allowing either a `number` or a `string` so as to keep the diff minimal in the tests which have `createComment` and `createReply` helpers. This also makes it more agnostic to what the API sends.

**Pull request merged**